server/routes/bookedtoday/arrivals/autoMatchingRecords/index.ts

```diff
@@ -10,15 +10,18 @@ import validate from '../../../../middleware/validationMiddleware'
 import MatchedRecordSelectionValidation from '../searchforexisting/validation/matchedRecordSelectionValidation'
 import Routes from '../../../../utils/routeBuilder'
 
+import * as backTrackPrevention from '../../../../middleware/backTrackPreventionMiddleware'
+
 export default function routes(services: Services): Router {
   const singleMatchController = new SingleMatchingRecordFoundController(services.expectedArrivalsService)
   const noMatchController = new NoMatchingRecordsFoundController(services.expectedArrivalsService)
   const mulitipleMatchController = new MultipleMatchingRecordsFoundController(services.expectedArrivalsService)
+  const checkIsLocked = backTrackPrevention.isLocked(services.lockManager, '/duplicate-booking-prevention')
 
   return Routes.forRole(Role.PRISON_RECEPTION)
-    .get('/prisoners/:id/record-found', singleMatchController.view())
-    .get('/prisoners/:id/no-record-found', noMatchController.view())
-    .get('/prisoners/:id/possible-records-found', mulitipleMatchController.view())
+    .get('/prisoners/:id/record-found', checkIsLocked, singleMatchController.view())
+    .get('/prisoners/:id/no-record-found', checkIsLocked, noMatchController.view())
+    .get('/prisoners/:id/possible-records-found', checkIsLocked, mulitipleMatchController.view())
     .post(
       '/prisoners/:id/possible-records-found',
       validate(MatchedRecordSelectionValidation),
```

Comment:
would it make sense to add it to the POST too?

Reply:
I did think about post but didn't think it was needed. User would need to be on the page to do the post.

Reply:
The lock could be created whilst they are on the page before clicking submit, so would need to protect this if the POST was doing anything significant side effects/writing data/making api calls etc... 

I think it's fine in this case - the main POST is protected and the other POSTs only modify session state